src/Adapters/Logger/WinstonLogger.js

```diff
@@ -93,25 +91,23 @@ export function configureLogger({
     options.json = true;
     options.stringify = true;
   }
-  updateTransports(options);
+  configureTransports(options);
 }
 
 export function addTransport(transport) {
-  additionalTransports.push(transport);
-  updateTransports();
+  logger.add(transport);
 }
 
 export function removeTransport(transport) {
-  const transportName =
-    typeof transport == 'string' ? transport : transport.name;
-  const transports = Object.assign({}, logger.transports);
-  delete transports[transportName];
-  logger.configure({
-    transports: _.values(transports),
-  });
-  _.remove(additionalTransports, transport => {
-    return transport.name === transportName;
+  const matchingTransport = logger.transports.find(t1 => {
```

Comment:
Yes I think you are right, the code has changed so now it might be possible to have two transports with the same name.  

I think previously this code prevented the above scenario.

```
 additionalTransports.forEach(transport => {
    transports[transport.name] = transport;
  });
```

I'll investigate...